src/landscape.cpp

```diff
@@ -1028,7 +1030,7 @@ static bool FindSpring(TileIndex tile, void *)
  */
 static bool MakeLake(TileIndex tile, void *user_data)
 {
-	uint height_lake = *(uint*)user_data;
+	uint height_lake = *(int*)user_data;
```

Comment:
Why?

Reply:
Declaration of `IsTileFlat(TileIndex tile, int *h = nullptr);` requires passing `h` as `int`.

On `FlowRiver` I changed the way the height of the lake is retrieved by using solely `IsTileFlat` for performance reasons, instead of `TileHeight` + `IsTileFlat`.
```
int height_lake;
IsTileFlat(lake_centre, &height_lake);
CircularTileSearch(&lake_centre, range, MakeLake, &height_lake);
```
On the `MakeLake` callback, I am unwrapping the value as it was passed, as `int`, but to make an equality comparison with `TileHeight`, it needs to be `uint` again.

I've already prepared changes to `MakeLake` in another PR, which also switched to using `IsTileFlat`, If both PRs are merged, they're all going to be `int`.

Reply:
I guess you have not spotted why I said "why": `height_lake` is declared `uint`, but you're dereferencing `user_data` as `int`. 

So effectively you are still passing `uint` to `IsTileFlat`. Having said that, rabbit hole and all, `IsTileFlat`'s height should probably be just `uint`. As that's the type `TileHeight` returns.